### PR TITLE
Fix inaccurate fee estimations

### LINKED ISSUE #4777: Fee set in Leather is not what it is transmitted with
The user reported selecting medium or high, they were likely 34 low, 36 mid, 38 sats/vB high. Average fees went to 45 quickly then 60 and back to 40 again. 

image

Looks like this one that was fixed recently: https://github.com/leather-wallet/extension/issues/4541

### PR DESCRIPTION
**Pull request by kyranjamie**
> Try out Leather build aab07ae — [Extension build](https://github.com/leather-wallet/extension/actions/runs/8829222346), [Test report](https://leather-wallet.github.io/playwright-reports/fix/fee-estimations), [Storybook](https://fix-fee-estimations--65982789c7e2278518f189e7.chromatic.com), [Chromatic](https://www.chromatic.com/library?appId=65982789c7e2278518f189e7&branch=fix/fee-estimations)

This PR fixes the [issues relating to fee estimations describe in Notion](https://www.notion.so/trustmachines/Fee-underestimation-research-69b7e4d21ba74ecba12b426b6be30fb4?pvs=4).

The previously calculation lead to a situation whereby some estimations would be counting one fewer input than actually used, thus vastly decreasing the fee rate.

Open to feedback how we can extend the tests of the coin selection. 

@fbwoolf can you please QA this to double check

https://github.com/leather-wallet/extension/assets/1618764/a604acca-8513-47a0-ab32-3662a5392f1d

### CODE REVIEW

**Comment by fbwoolf** [comment]
cc @314159265359879 to help QA here?

File: src/app/common/transactions/bitcoin/coinselect/local-coin-selection.spec.ts [hunk 1]
```diff
@@ -21,25 +21,74 @@ const demoUtxos = [
...
+      expect(estimation.txVBytes).toBeGreaterThan(750);
+      expect(estimation.txVBytes).toBeLessThan(751);
+    });
+  });
+
+  describe('sorting algorithm', () => {
     test('that it filters out dust utxos', () => {
-      const result = generate10kSpendWithTestData('tb1qt28eagxcl9gvhq2rpj5slg7dwgxae2dn2hk93m');
+      const result = generate10kSpendWithDummyUtxoSet('tb1qt28eagxcl9gvhq2rpj5slg7dwgxae2dn2hk93m');
+      console.log(result);
```
Comment (fbwoolf) [inline src/app/common/transactions/bitcoin/coinselect/local-coin-selection.spec.ts:85]: Remove?

File: src/app/pages/send/send-crypto-asset-form/form/btc/btc-send-form-confirmation.tsx [hunk 2]
```diff
@@ -50,13 +51,52 @@ export function BtcSendFormConfirmation() {
   const navigate = useNavigate();
   const { tx, recipient, fee, arrivesIn, feeRowValue } = useBtcSendFormConfirmationState();
 
+  const transaction = useMemo(() => btc.Transaction.fromRaw(hexToBytes(tx)), [tx]);
+  // const inputs = useMemo(() => getPsbtTxInputs(transaction), [transaction]);
```
Comment (fbwoolf) [inline src/app/pages/send/send-crypto-asset-form/form/btc/btc-send-form-confirmation.tsx:55]: Remove commented code?

**Review by fbwoolf — COMMENTED** [review]
Code looks good here, and I'm not running into any bugs testing it. 👍

**Comment by alter-eggo** [comment]
@kyranjamie I believe we need to fix calc in `selectTaprootInscriptionTransferCoins` as well?

**Comment by alter-eggo** [comment]
checked btc send with 3ns inputs and 2ns outputs and it worked for me 👍

**Comment by kyranjamie** [comment]
> @kyranjamie I believe we need to fix calc in selectTaprootInscriptionTransferCoins as well? 

Eurgh, this needs refactoring. Though I think it doesn't have same problem, as the other with the for loop.

File: src/app/common/transactions/bitcoin/utils.ts [hunk 3]
```diff
@@ -80,22 +80,22 @@ export function filterUneconomicalUtxos({
...
 
   const txSizer = new BtcSizeFeeEstimator();
   const sizeInfo = txSizer.calcTxSize({
     // Only p2wpkh is supported by the wallet
     input_script: 'p2wpkh',
-    input_count: inputLength,
+    input_count: inputCount,
     // From the address of the recipient, we infer the output type
-    [outputAddressTypeWithFallback + '_output_count']: outputLength,
+    [outputAddressTypeWithFallback + '_output_count']: outputCount,
```
Comment (alter-eggo) [inline src/app/common/transactions/bitcoin/utils.ts:98]: as I understand we can skip adding `_outputs_counts` for different output types? 🤔
like it is now in `getSizeInfoMultipleRecipients`

**Review by fbwoolf — APPROVED** [review]

File: src/app/common/transactions/bitcoin/utils.ts (hunk 3 above)
Comment (kyranjamie) [reply src/app/common/transactions/bitcoin/utils.ts:98]: Not entirely sure what you mean by this @alter-eggo ?

Comment (kyranjamie) [reply src/app/common/transactions/bitcoin/utils.ts:98]: You do this as well in the multiple recipients code ` acc[type + '_output_count'] = count;`

Comment (kyranjamie) [reply src/app/common/transactions/bitcoin/utils.ts:98]: Open for changes if needed here, but merging to not hold up fixes

**Pull request merged**